scripts/deploy-tarball.js

```diff
@@ -0,0 +1,155 @@
+// @ts-check
+const path = require('path')
+const execa = require('execa')
+const fs = require('fs/promises')
+
+const cwd = process.cwd()
+
+async function main() {
+  const deployDir = path.join(cwd, 'files')
+  const publicDir = path.join(deployDir, 'public')
+  await fs.mkdir(publicDir, { recursive: true })
+  await fs.writeFile(
+    path.join(deployDir, 'package.json'),
+    JSON.stringify({
+      name: 'files',
+      dependencies: {},
+      scripts: {
+        build: 'node inject-deploy-url.js',
+      },
+    })
+  )
+  await fs.copyFile(
+    path.join(cwd, 'scripts/inject-deploy-url.js'),
+    path.join(deployDir, 'inject-deploy-url.js')
+  )
+
+  let nativePackagesDir = path.join(cwd, 'packages/next-swc/crates/napi/npm')
+  let platforms = (await fs.readdir(nativePackagesDir)).filter(
+    (name) => !name.startsWith('.')
+  )
+
+  const optionalDeps = {}
+  const { version } = JSON.parse(
+    await fs.readFile(path.join(cwd, 'lerna.json'), 'utf8')
+  )
+
+  await Promise.all(
+    platforms.map(async (platform) => {
+      let binaryName = `next-swc.${platform}.node`
+      await fs.cp(
+        path.join(cwd, 'packages/next-swc/native', binaryName),
+        path.join(nativePackagesDir, platform, binaryName)
+      )
+      let pkg = JSON.parse(
+        await fs.readFile(
+          path.join(nativePackagesDir, platform, 'package.json'),
+          'utf8'
+        )
+      )
+      pkg.version = version
+      await fs.writeFile(
+        path.join(nativePackagesDir, platform, 'package.json'),
+        JSON.stringify(pkg, null, 2)
+      )
+      const { stdout } = await execa(`npm`, [
+        `pack`,
+        `${path.join(nativePackagesDir, platform)}`,
+      ])
```

Comment:
Let's add `COREPACK_ENABLE_STRICT: '0'` to the env here

Reply:
Why is this needed? This handling was copied from our actual publish script

Reply:
It would only be needed if `corepack enable` was run or if corepack becomes default enabled in a future version of Node.js